packages/core/src/next/plugin.ts

```diff
@@ -96,6 +96,9 @@ async function boostrapCodegen(port: number, path: string) {
         baseDirectory + '/**/*.{ts,tsx}',
         baseDirectory + '/types/**/*.ts',
       ],
+      hooks: {
+        afterOneFileWrite: ['prettier --write'],
```

Comment:
This is going to crash if `prettier` does not exist, no?

Reply:
1. Yes, you're right. It will in fact crash
2. The [docs](https://the-guild.dev/graphql/codegen/docs/config-reference/lifecycle-hooks#afteronefilewrite) state that `afterOneFileWrite` is the recommend way since it's only run for a single file when that changes
3. I'm pretty sure that prettier resolves this by default for themselves

Reply:
> I'm pretty sure that prettier resolves this by default for themselves

That's not what I'm saying, this point connects to 1, where either we introduce this as an option where the user has to opt-in voluntarily or we detect prettier. The reasoning here is that a lot of folks might just check this into their `.prettierignore` or might not have gotten prettier installed in the first place. The presence of `prettier` and by extension the willingness to subject generated files to these rules shouldn't be a strict opinion to code-generation working.

Reply:
Okay that makes sense. Detecting prettier doesn't seem that trivial since it could be either
1. installed globally (easiest)
2. Installed in the current package
3. Installed in the workspace root

But even if we decide to add an option here we still need to figure out how to run the prettier binary since this changes depending on the setup

Reply:
I think this is what you meant

Reply:
The running of it doesn't change depending on setup tbh, just prettier is enough if they enable it imho

Reply:
I have two ideas right now. We could either:

1. Give the user the option to add the hooks themselves (my preferred option; would allow them to run their linter as well. Not only prettier.)
2. Just pass in a flag which will run prettier.


What do you think?

Reply:
We could give them the option, there are a few places where we manually write files through `fs` but those are one-time writes so we could just stop writing when they are already in the filesystem so it is only a one time git diff for the user.